cpp/src/core/sroptcryst.h

```diff
@@ -927,14 +908,35 @@ class srTOptCryst : public srTGenOptElem {
 		// Calculate direction cosine gamma0, reflection asymmetry parameter bee, 
 		// deviation parameter Adev and normalized deviation parameter zeeC (as 
 		// defined by Zachariasen gamma0, b, alpha and z.) 
-		double gamma0 = -u0X[1];
+		double gamma0;
+            
+		if(m_ug == 1)
+		{
+			gamma0 = -u0X[1];
+		}
+		else if (m_ug == 2)
+		{
+			gamma0 = -u0X[1];        
+		}
```

Comment:
Left the previous version (the suggested code seems to be equivalent to it; maybe some changes were made since the original suggestion?).